MonoGame.Framework/FrameworkDispatcher.cs

```diff
@@ -2,15 +2,57 @@
 // This file is subject to the terms and conditions defined in
 // file 'LICENSE.txt', which is part of this source code package.
 
+using System;
+using Microsoft.Xna.Framework.Audio;
+
 namespace Microsoft.Xna.Framework
 {
-	public static class FrameworkDispatcher
-	{
-		public static void Update()
-		{
-			// "Updates the status of various framework components
-			// (such as power state and media), and raises related events"
-		}
-	}
+    /// <summary>
+    /// Helper class for processing internal framework events.
+    /// </summary>
+    /// <remarks>
+    /// If you use <see cref="Game"/> class, <see cref="Update()"/> is called automatically.
+    /// Otherwise you must call it as part of your game loop.
+    /// </remarks>
+    public static class FrameworkDispatcher
+    {
+        private static bool _initialized = false;
+
+        /// <summary>
+        /// Processes framework events.
+        /// </summary>
+        public static void Update()
+        {
+            if (!_initialized)
+                Initialize();
+
+            DoUpdate();
+        }
+
+        private static void DoUpdate()
+        {
+            DynamicSoundEffectInstanceManager.UpdatePlayingInstances();
+            SoundEffectInstancePool.Update();
+        }
+
+        private static void Initialize()
+        {
+            // Initialize sound system
+#if DIRECTX
+            SoundEffect.InitializeSoundEffect();
+#elif OPENGL
+            try
+            {
+                var oal = OpenALSoundController.GetInstance;
+            }
+            catch (DllNotFoundException ex)
+            {
+                throw new NoAudioHardwareException("Failed to init OpenALSoundController", ex);
+            }
+#endif

```

Comment:
We should avoid the `#if`s here as it makes it hard to extend things to the console platforms.

Still why is this here?  Usually if one were implementing some code that didn't use `Game` yet used `SoundEffect`...  it would go something like this:

```
var sfx = SoundEffect.FromStream(stream);
sfx.Play();
FrameworkDispatcher.Update();
```

Essentially initializing things a second time no?